### PR TITLE
update documentation for ssh keys

### LINKED ISSUE #58: Improve documentation about ssh keys
Based on the recent question on the mailing list: https://groups.google.com/forum/?utm_medium=email&utm_source=footer#!msg/ros-sig-buildfarm/P32OdoGFYkg/awEAi8Bx4jgJ

The description of the `ssh_keys` for all three machine types is pretty confusing.
I would recommend something like this:

```
 * Configure as many public ssh keys as you want for administrators to log in.
 * On the repo machine make sure there is public ssh key which matches the private ssh key `jenkins::private_ssh_key` provisioned on the master.
```

The key `jenkins-slave::authorized_keys` described for the repo machine seems to not exist in the config repository.
Is this a private ssh key? If so the documentation should state that explicitly.

Also the indentation of the configuration option for the repo machine seem to be wrong.

### PR DESCRIPTION
**Pull request by tfoote**
Fixes #58

A little more info on ssh keys.

### CODE REVIEW

**Comment by esteve** [comment]
+1

File: README.md [hunk 1]
```diff
@@ -84,8 +84,9 @@ On all three:
 * `autoreconfigure::branch`
 * If you are forking into a repo and using a different branch name, update the autoreconfigure command to point to the right branch.
 * `ssh_keys`
- * Configure as many license keys as you want for administrators to log in.
- * Make sure there is at least one key for jenkins-slave on the repo matching the `jenkins::private_key` provisioned on the master.
+ * Configure as many public ssh-keys as you want for administrators to log in. It's recommended at least one for root.
```
Comment (dirk-thomas) [inline README.md]: Please stick to the "not more than one sentence per line" rule.

File: README.md [hunk 2]
```diff
@@ -84,8 +84,9 @@ On all three:
...
+ * On the repo machine make sure there is at least one key for the jenkins-slave user matching the
+  `jenkins::private_key` provisioned on the master.
```
Comment (dirk-thomas) [inline README.md]: This is still a bit fuzzy. It doesn't clarify what are public and what are private keys. I would suggest to include that as or similar to the phrase in the referenced issue.

File: README.md [hunk 3]
```diff
@@ -133,7 +134,9 @@ On the master:
 * This is the string contents for the authorized keys for the slaves to push into the repo.
   It should match the `jenkins::private_ssh_key` on the master.
   * `jenkins::private_ssh_key`
-  * The key which authorizes access to push content into the repository or to connect back to the master from a job.
+  * The ssh private key which authorizes access to push content into the repository or to connect back to the master from a job.
```
Comment (dirk-thomas) [inline README.md]: I am not sure anymore but is it currently being used to "connect back to the master from a job"?

**Comment by dirk-thomas** [comment]
May be adding a sentence to clarify how the ssh keys are usually obtained would help users. E.g. for the private keys: "extract them from the user X on machine Y".

**Comment by tfoote** [comment]
Gave more clarity on private vs public. I don't want to get down in the weeds about how to create/obtain ssh keys.

**Comment by dirk-thomas** [comment]
I think a sentence about which ssh key is supposed to be used would help the users, e.g. the one asking the recent question on the mailing list. Currently at least to me it is unclear from which user on the repo machine the private key is required to be on the master.

**Comment by tfoote** [comment]
@dirk-thomas How can I make this clearer? https://github.com/ros-infrastructure/buildfarm_deployment/pull/59/files#diff-04c6e90faac2675aa89e2176d2eec7d8R89

**Comment by dirk-thomas** [comment]
I missed that it is in the "On all three" section - I only looked at the "Repo" section. That is absolutely sufficient. Sorry for the noise.

+1

**Pull request merged**